apps/performance-tests/src/unified-selection/HiliteSet.test.ts

```diff
@@ -0,0 +1,117 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import { SnapshotDb } from "@itwin/core-backend";
+import { SchemaContext, SchemaJsonLocater } from "@itwin/ecschema-metadata";
+import { createECSchemaProvider, createECSqlQueryExecutor } from "@itwin/presentation-core-interop";
+import { createHiliteSetProvider, Selectable, Selectables } from "@itwin/unified-selection";
+import { Datasets, IModelName } from "../util/Datasets";
+import { run, RunOptions } from "../util/TestUtilities";
+
+describe("hilite", () => {
+  const additionalElementCount = 5;
+  const additionalFunctionalElement2dCount = 1;
+
+  function createSelection({ fullClassName, start, count }: { fullClassName: string; start: number; count: number }) {
+    const selectables: Selectable[] = [];
+    for (let i = start; i < start + count + additionalElementCount; i++) {
+      selectables.push({ className: fullClassName, id: `0x${i.toString(16)}` });
+    }
+    return Selectables.create(selectables);
+  }
+
+  runHiliteTest({
+    testName: "50k elements",
+    iModelName: "50k nested elements",
+    selection: createSelection({ fullClassName: "Generic:PhysicalObject", start: 14, count: 50_000 }),
+    expectedCounts: { elements: 50000 },
+  });
```

Comment:
I don't understand this. As far as I can see, the `50k nested elements` iModel has 25k 2d and 25k 3d elements. 
- The input (selection) is 50k 3d elements - what elements are they, if the iModel only contains 25k?
- How do we get 50k elements in the results if the iModel only has 25k 3d elements?

Reply:
That's because the class name is only relevant to determine what the type of the element is, in this case 50k elements were passed as `Generic:PhysicalObject` but the query was treating it as `BisCore:Element` so the 25k 2D elements were also included. This was intended except for the fact that ``BisCore:Element` was supposed to be used instead of `Generic:PhysicalObject`. Fixed.